conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureBodySerDe.java

```diff
@@ -100,7 +101,7 @@ public void serialize(BinaryResponseBody value, HttpServerExchange exchange) thr
     @Override
     public InputStream deserializeInputStream(HttpServerExchange exchange) {
         String contentType = getContentType(exchange);
-        if (!contentType.startsWith(BINARY_CONTENT_TYPE)) {
+        if (!contentType.toLowerCase(Locale.ROOT).startsWith(BINARY_CONTENT_TYPE)) {
```

Comment:
should we push the toLowerCase into `getContentType`?

Reply:
Sounds reasonable to me

Reply:
Actually we could do this without allocating

```suggestion
        if (contentType.length() < BINARY_CONTENT_TYPE.length()
                || !contentType.regionMatches(/* ignoreCase= */ true, 0, BINARY_CONTENT_TYPE, 0, BINARY_CONTENT_TYPE.length())) {
```

Reply:
Both are good suggestions! Went with the `contentType#regionMatches` approach for now given that we only have a single place where we do this comparison (all other paths are handled in the respective encodings). Can pull this into a helper method once we have more usage.